src/main/java/pl/taskyers/taskybase/project/slo/ProjectInvitationSLOImpl.java

```diff
@@ -0,0 +1,85 @@
+package pl.taskyers.taskybase.project.slo;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import pl.taskyers.taskybase.core.emails.EmailConstants;
+import pl.taskyers.taskybase.core.emails.EmailSLO;
+import pl.taskyers.taskybase.core.messages.MessageCode;
+import pl.taskyers.taskybase.core.messages.MessageType;
+import pl.taskyers.taskybase.core.messages.ResponseMessage;
+import pl.taskyers.taskybase.core.roles.slo.RoleSLO;
+import pl.taskyers.taskybase.core.slo.TokenSLO;
+import pl.taskyers.taskybase.core.users.converters.AccountConverter;
+import pl.taskyers.taskybase.core.users.entity.UserEntity;
+import pl.taskyers.taskybase.core.users.slo.UserSLO;
+import pl.taskyers.taskybase.project.entity.ProjectEntity;
+import pl.taskyers.taskybase.project.entity.ProjectInvitationTokenEntity;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class ProjectInvitationSLOImpl implements ProjectInvitationSLO {
+    
+    private final UserSLO userSLO;
+    
+    private final EmailSLO emailSLO;
+    
+    private final ProjectSLO projectSLO;
+    
+    private final RoleSLO roleSLO;
+    
+    private final TokenSLO projectInvitationTokenSLO;
+    
+    @Override
+    public ResponseEntity sendEmailWithInvitationToken(String userName, String projectName) {
+        if ( userSLO.getEntityByUsername(userName).isPresent() && projectSLO.getProjectEntityByName(projectName).isPresent() ) {
+            UserEntity userEntity = userSLO.getEntityByUsername(userName).get();
+            ProjectEntity projectEntity = projectSLO.getProjectEntityByName(projectName).get();
+            if ( roleSLO.hasPermission(userEntity, projectEntity, "project.invite.others") ) {
```

Comment:
Tutaj bym dał pole z tego interfejsu zamiast stringa z rolą, po to tam dodajemy zeby tutaj nie pisac tych stringow :)